### PR TITLE
Change person classes to patient classes and add information fields for GUI

### PR DESCRIPTION
**Pull request by guokweijie**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -40,13 +40,13 @@ Contact number: 91234567
 Appointment Date: 5/10/22
 ````
 
-### Editing a person: `edit`
+### Editing a patient: `edit`
 
 Edit a patient’s information, such as age, contact number, appointment date and doctor’s notes.
 
 Format: `edit Person Title: content`
```
Comment (TZL0) [inline docs/UserGuide.md:47]: patient

File: docs/tutorials/AddRemark.md [hunk 2]
```diff
@@ -240,7 +240,7 @@ Let’s change `RemarkCommand` and `RemarkCommandParser` to use the new `Remark`
 
 ## Add a placeholder element for remark to the UI
 
-Without getting too deep into `fxml`, let’s go on a 5 minute adventure to get some placeholder text to show up for each person.
+Without getting too deep into `fxml`, let’s go on a 5 minute adventure to get some placeholder text to show up for each patient.
 
 Simply add the following to [`seedu.address.ui.PersonCard`](https://github.com/se-edu/addressbook-level3/commit/850b78879582f38accb05dd20c245963c65ea599#diff-639834f1e05afe2276a86372adf0fe5f69314642c2d93cfa543d614ce5a76688).
```
Comment (TZL0) [inline docs/tutorials/AddRemark.md:245]: patient

File: docs/tutorials/AddRemark.md [hunk 3]
```diff
@@ -311,9 +311,9 @@ Just add [this one line of code!](https://github.com/se-edu/addressbook-level3/c
 **`PersonCard.java`:**
```
Comment (TZL0) [inline docs/tutorials/AddRemark.md:311]: Patient

File: docs/tutorials/AddRemark.md [hunk 4]
```diff
@@ -343,25 +343,25 @@ save it with `Model#setPerson()`.
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
```
Comment (TZL0) [inline docs/tutorials/AddRemark.md:343]: PATIENT

File: docs/tutorials/AddRemark.md [hunk 5]
```diff
@@ -343,25 +343,25 @@ save it with `Model#setPerson()`.
...
         }
 
-        Person personToEdit = lastShownList.get(index.getZeroBased());
-        Person editedPerson = new Person(
-                personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
-                personToEdit.getAddress(), remark, personToEdit.getTags());
+        Person patientToEdit = lastShownList.get(index.getZeroBased());
+        Person editedPatient = new Person(
+                patientToEdit.getName(), patientToEdit.getPhone(), patientToEdit.getEmail(),
+                patientToEdit.getAddress(), remark, patientToEdit.getTags());
```
Comment (TZL0) [inline docs/tutorials/AddRemark.md:350]: Patient

File: docs/tutorials/AddRemark.md [hunk 6]
```diff
@@ -343,25 +343,25 @@ save it with `Model#setPerson()`.
...
 
-        model.setPerson(personToEdit, editedPerson);
+        model.setPerson(patientToEdit, editedPatient);
         model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
```
Comment (TZL0) [inline docs/tutorials/AddRemark.md:353]: Patient

File: docs/tutorials/AddRemark.md [hunk 7]
```diff
@@ -343,25 +343,25 @@ save it with `Model#setPerson()`.
...
     }
 
     /**
      * Generates a command execution success message based on whether
      * the remark is added to or removed from
-     * {@code personToEdit}.
+     * {@code patientToEdit}.
      */
-    private String generateSuccessMessage(Person personToEdit) {
+    private String generateSuccessMessage(Person patientToEdit) {
```
Comment (TZL0) [inline docs/tutorials/AddRemark.md:362]: Patient

File: docs/tutorials/RemovingFields.md [hunk 8]
```diff
@@ -28,7 +28,7 @@ IntelliJ IDEA provides a refactoring tool that can identify *most* parts of a re
 
 ### Assisted refactoring
 
-The `address` field in `Person` is actually an instance of the `seedu.address.model.person.Address` class. Since removing the `Address` class will break the application, we start by identifying `Address`'s usages. This allows us to see code that depends on `Address` to function properly and edit them on a case-by-case basis. Right-click the `Address` class and select `Refactor` \> `Safe Delete` through the menu.
+The `address` field in `Person` is actually an instance of the `seedu.address.model.patient.Address` class. Since removing the `Address` class will break the application, we start by identifying `Address`'s usages. This allows us to see code that depends on `Address` to function properly and edit them on a case-by-case basis. Right-click the `Address` class and select `Refactor` \> `Safe Delete` through the menu.
```
Comment (TZL0) [inline docs/tutorials/RemovingFields.md:31]: Patient

File: docs/tutorials/RemovingFields.md [hunk 9]
```diff
@@ -100,7 +100,7 @@ In `src/test/data/`, data meant for testing purposes are stored. While keeping t
 
 ```json
 {
-  "persons": [ {
+  "patients": [ {
     "name": "Person with invalid name field: Ha!ns Mu@ster",
```
Comment (TZL0) [inline docs/tutorials/RemovingFields.md:104]: Patient

File: docs/tutorials/TracingCode.md [hunk 10]
```diff
@@ -189,22 +189,22 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
    @Override
    public CommandResult execute(Model model) throws CommandException {
        ...
-       Person personToEdit = lastShownList.get(index.getZeroBased());
-       Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
-       if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+       Person patientToEdit = lastShownList.get(index.getZeroBased());
+       Person editedPatient = createEditedPerson(patientToEdit, editPersonDescriptor);
+       if (!patientToEdit.isSamePerson(editedPatient) && model.hasPerson(editedPatient)) {
            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
```
Comment (TZL0) [inline docs/tutorials/TracingCode.md:195]: Patient

File: docs/tutorials/TracingCode.md [hunk 11]
```diff
@@ -189,22 +189,22 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
...
        }
-       model.setPerson(personToEdit, editedPerson);
+       model.setPerson(patientToEdit, editedPatient);
        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
-       return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, editedPerson));
+       return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, editedPatient));
    }
```
Comment (TZL0) [inline docs/tutorials/TracingCode.md:200]: Patient

File: docs/tutorials/TracingCode.md [hunk 12]
```diff
@@ -189,22 +189,22 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
...
-     To provide some context, given below is the class diagram of the `Model` component. See if you can figure out where the 'filtered list' of persons is being tracked.
+   * it uses the `setPerson()` method (defined in the interface `Model` and implemented in `ModelManager` as per the usual pattern) to update the patient data.
+   * it uses the `updateFilteredPersonList` method to ask the `Model` to populate the 'filtered list' with _all_ patients.<br>
+     FYI, The 'filtered list' is the list of patients resulting from the most recent operation that will be shown to the user immediately after. For the `edit` command, we populate it with all the patients so that the user can see the edited patient along with all other patients. If this was a `find` command, we would be setting that list to contain the search results instead.<br>
+     To provide some context, given below is the class diagram of the `Model` component. See if you can figure out where the 'filtered list' of patients is being tracked.
      <img src="../images/ModelClassDiagram.png" width="450" /><br>
```
Comment (TZL0) [inline docs/tutorials/TracingCode.md:208]: Patient

File: docs/tutorials/TracingCode.md [hunk 13]
```diff
@@ -231,7 +231,7 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
      * {@code JsonSerializableAddressBook}.
      */
     public JsonSerializableAddressBook(ReadOnlyAddressBook source) {
-        persons.addAll(
+        patients.addAll(
             source.getPersonList()
```
Comment (TZL0) [inline docs/tutorials/TracingCode.md:235]: Patient

**Review by TZL0 — CHANGES_REQUESTED** [review]
Hi Wei Jie, I found several 'persons' to be renamed

**Review by TZL0 — APPROVED** [review]

**Review by SpecOps2016 — APPROVED** [review]
LGTM

**Pull request merged**